foxpuppet/windows/browser/notifications/base.py

```diff
@@ -0,0 +1,46 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from abc import ABCMeta
+
+from foxpuppet.region import Region
+
+
+class BaseNotification(Region):
+    """Abstract base class for any kind of notification."""
+
+    __metaclass__ = ABCMeta
+
+    @staticmethod
+    def create(window, root):
+        notifications = {}
+        _id = root.get_attribute('id')
```

Comment:
I would propose to use `get_property()` here to fetch the id. Not sure if it is exposed as DOM attribute at any time.